src/lesson18/stream/api/MainLambdaTemplateStudents.java

```diff
@@ -0,0 +1,127 @@
+package lesson18.stream.api;
+
+import java.time.LocalDate;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.Comparator;
+import java.lang.Comparable;
+
+import static lesson18.stream.api.FruitType.*;
+import static lesson18.stream.api.Vitamin.A;
+import static lesson18.stream.api.Vitamin.B;
+import static lesson18.stream.api.Vitamin.B1;
+import static lesson18.stream.api.Vitamin.B2;
+import static lesson18.stream.api.Vitamin.C;
+import static lesson18.stream.api.Vitamin.D;
+import static lesson18.stream.api.Vitamin.E;
+import static lesson18.stream.api.Vitamin.P;
+
+/**
+ * @author spasko
+ */
+public class MainLambdaTemplateStudents implements Comparable<Fruit> {
+    public static void main(String[] args) {
+        List<Fruit> fruits = fillFruitsList();
+//        fruits.forEach(System.out::println);
+
+        System.out.println("Part 1");
+
+        fruits.stream()
+                .peek(fr -> System.out.println(fr))
+                .flatMap(fr -> fr.getVitamins()
```

Comment:
In order to avoid iterating over all vitamins lists you can use e.g. fruit.getVitamins().contains(A)